src/main/java/com/exadel/discountwebapp/filter/SearchOperation.java

```diff
@@ -22,6 +20,6 @@ public static SearchOperation getOperation(String input) {
         return Arrays.stream(values())
                 .filter(e -> e.operation.equals(input))
                 .findFirst()
-                .orElseThrow(() -> new NotAllowedOperationException(String.format("Operation [%s] is not allowed", input)));
+                .orElse(EQUALITY);
```

Comment:
why not to throw exception?
if there is no valid operation then the request is malformed

Reply:
I returned that back. Our regexp skips the operations that are not allowed and that exception will never be used